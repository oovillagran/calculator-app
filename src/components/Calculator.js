const Calculator = () => (
  <div className="calculator-container">
    <h2 className="calculator-title">Calculator App</h2>
    <ChildComponent />
  </div>
);

function ChildComponent() {
  return (
    <div className="main-container">
      <input className="input-screen" type="text" id="input" name="name" pattern="[0-9+\-*/()\s]+" />
      <div className="buttons-container">
        <button className="button math-operation" type="button">AC</button>
        <button className="button math-operation" type="button">+/-</button>
        <button className="button math-operation" type="button">%</button>
        <button className="button math-operation orange" type="button">+</button>
        <br />
        <button className="button math-number" type="button">7</button>
        <button className="button math-number" type="button">8</button>
        <button className="button math-number" type="button">9</button>
        <button className="button math-operation orange" type="button">x</button>
        <br />
        <button className="button math-number" type="button">4</button>
        <button className="button math-number" type="button">5</button>
        <button className="button math-number" type="button">6</button>
        <button className="button math-operation orange" type="button">-</button>
        <br />
        <button className="button math-number" type="button">1</button>
        <button className="button math-number" type="button">2</button>
        <button className="button math-number" type="button">3</button>
        <button className="button math-operation orange" type="button">+</button>
        <br />
        <button className="button math-number" type="button">0</button>
        <button className="button math-number" type="button">.</button>
        <button className="button math-operation orange" type="button">=</button>
      </div>
    </div>
  );
}

export default Calculator;
