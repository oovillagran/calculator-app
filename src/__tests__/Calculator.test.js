import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  test('matches snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('displays numbers when buttons are clicked', () => {
    const { getByText, getByPlaceholderText } = render(<Calculator />);
    const input = getByPlaceholderText('0');
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('3'));
    expect(input.value).toBe('123');
  });

  test('adds two numbers correctly', () => {
    const { getByText, getByPlaceholderText } = render(<Calculator />);
    const input = getByPlaceholderText('0');
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(input.value).toBe('3');
  });

  test('subtracts two numbers correctly', () => {
    const { getByText, getByPlaceholderText } = render(<Calculator />);
    const input = getByPlaceholderText('0');
    fireEvent.click(getByText('5'));
    fireEvent.click(getByText('-'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(input.value).toBe('3');
  });

  test('multiplies two numbers correctly', () => {
    const { getByText, getByPlaceholderText } = render(<Calculator />);
    const input = getByPlaceholderText('0');
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('x'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));
    expect(input.value).toBe('6');
  });

  test('divides two numbers correctly', () => {
    const { getByText, getByPlaceholderText } = render(<Calculator />);
    const input = getByPlaceholderText('0');
    fireEvent.click(getByText('9'));
    fireEvent.click(getByText('รท'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));
    expect(input.value).toBe('3');
  });
});
