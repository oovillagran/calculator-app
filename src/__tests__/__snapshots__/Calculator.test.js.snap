// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component matches snapshot 1`] = `
<div
  class="calculator-container"
>
  <h2
    class="calculator-title"
    style="color: rgb(255, 255, 255);"
  >
    Calculator App
  </h2>
  <div
    class="main-container"
  >
    <div
      class="calculator"
    >
      <input
        class="input-screen"
        id="input"
        name="name"
        pattern="[0-9+\\\\-*/()\\\\s]+"
        placeholder="0"
        readonly=""
        type="text"
        value="0"
      />
      <div
        class="button-panel buttons-container"
      >
        <button
          class="button height"
          style="background: rgb(239, 1, 7); color: rgb(255, 255, 255);"
          type="button"
        >
          AC
        </button>
        <button
          class="button height"
          type="button"
        >
          +/-
        </button>
        <button
          class="button height"
          type="button"
        >
          %
        </button>
        <button
          class="button orange height"
          type="button"
        >
          รท
        </button>
        <button
          class="button height"
          type="button"
        >
          7
        </button>
        <button
          class="button height"
          type="button"
        >
          8
        </button>
        <button
          class="button height"
          type="button"
        >
          9
        </button>
        <button
          class="button orange height"
          type="button"
        >
          x
        </button>
        <button
          class="button height"
          type="button"
        >
          4
        </button>
        <button
          class="button height"
          type="button"
        >
          5
        </button>
        <button
          class="button height"
          type="button"
        >
          6
        </button>
        <button
          class="button orange height"
          type="button"
        >
          -
        </button>
        <button
          class="button height"
          type="button"
        >
          1
        </button>
        <button
          class="button height"
          type="button"
        >
          2
        </button>
        <button
          class="button height"
          type="button"
        >
          3
        </button>
        <button
          class="button orange height"
          type="button"
        >
          +
        </button>
        <button
          class="button-zero height"
          type="button"
        >
          0
        </button>
        <button
          class="button height"
          type="button"
        >
          .
        </button>
        <button
          class="button orange height"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
